lib/net/http_fetch_plugin.js

```diff
@@ -140,7 +140,7 @@ shaka.net.HttpFetchPlugin = class {
           if (!readObj.done) {
             loaded += readObj.value.byteLength;
             if (streamDataCallback) {
-              streamDataCallback(readObj.value);
+              await streamDataCallback(readObj.value);
```

Comment:
I see the callback in StreamingEngine is async, so this makes sense.  But can you also please change the signature of the callback accordingly?

```js
/**
 * @param {?function(BufferSource):!Promise} streamDataCallback
 */
```

Reply:
Could you please do this in the entire codebase?  There are others in lib/util/, lib/net/, externs/, etc.  Thanks!